service/controller/v11/resource/pod/delete.go

```diff
@@ -38,80 +41,201 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		}
 	}
 
-	r.logger.LogCtx(ctx, "debug", "found the current version of the reconciled pod in the Kubernetes API")
+	{
+		a := currentPod.GetAnnotations()
+		if a == nil {
+			return microerror.Mask(missingAnnotationError)
```

Comment:
How to recover from that error?

Reply:
Don't know. This should actually never happen. When this happens we fucked up completely.

Reply:
If I understand correctly if this happens this is a bug. Let's add a comment that this should never happen or we have a bug.

Reply:
Done. Added some general godoc to the block and described the situation. Also added a TODO for the status magic to be improved.